examples/stm32f7/src/bin/usart_dma.rs

```diff
@@ -0,0 +1,46 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+#[path = "../example_common.rs"]
+mod example_common;
+use core::fmt::Write;
+use embassy::executor::Executor;
+use embassy::util::Forever;
+use embassy_stm32::dma::NoDma;
+use embassy_stm32::usart::{Config, Uart};
+use embassy_traits::uart::Write as _Write;
+use example_common::*;
+
+use cortex_m_rt::entry;
+use heapless::String;
+
+#[embassy::task]
+async fn main_task() {
```

Comment:
This example can be simplified by using the embassy:main macro eg https://github.com/embassy-rs/embassy/blob/4271226fc06130d30678739d4869db4fc1820fd8/examples/stm32f7/src/bin/blinky.rs#L14